packages/shared/src/components/profile/TopReaderWidget.tsx

```diff
@@ -0,0 +1,102 @@
+import { useQuery } from '@tanstack/react-query';
+import React, { ReactElement } from 'react';
+import { generateQueryKey, RequestKey, StaleTime } from '../../lib/query';
+import { fetchTopReaders } from '../../lib/topReader';
+import { disabledRefetch } from '../../lib/func';
+import { ActivityContainer, ActivitySectionHeader } from './ActivitySection';
+import { topReaderBadgeDocs } from '../../lib/constants';
+import { MedalBadgeIcon } from '../icons';
+import { IconSize } from '../Icon';
+import { BadgeIconGoldGradient } from '../badges/BadgeIcon';
+import {
+  Typography,
+  TypographyColor,
+  TypographyType,
+} from '../typography/Typography';
+import { formatDate, TimeFormatType } from '../../lib/dateFormat';
+import type { PublicProfile } from '../../lib/user';
+
+export const TopReaderWidget = ({
+  user,
+}: {
+  user: PublicProfile;
+}): ReactElement => {
+  const { data: topReader, isLoading: isTopReaderLoading } = useQuery({
```

Comment:
Do we use this query elsewhere?
If so maybe it's good to abstract it, just so we ensure the key to be the same everywhere etc.

Reply:
There's the one in `TopReaderbadgeModal`, it has the same key format

Reply:
Yeah so maybe let's refactor it to a simple hook.
You can look at `useSlack` for. instance (or any of those hooks)

It's just to ensure other devs don't mix only 1 key by accident.

Reply:
Yeah, I'll refactor it to a hook. Didn't do it because the userId was optional in the query, but Ante suggested to make it required, so works better as we can re-use it to a simple hook now